### PR TITLE
EOS-20657: Implement gpg verification for an ISO as alternative to hash based approach

### PR DESCRIPTION
**Pull request by dmitrydb**
I committed the base version of GPG authenticity validation for SW upgrade ISO

### CODE REVIEW

File: api/python/provisioner/api.py [hunk 1]
```diff
@@ -414,6 +414,34 @@ def get_swupgrade_info(release: str = None, iso_path: str = None,
...
+
+    Parameters
+    ----------
+    iso_path: str
+        Path to SW upgrade single ISO bundle
+    sig_file: str
+        Path to file with ISO signature
+    gpg_pub_key: str
+        (Optional) Path to the custom GPG public key
+    import_pub_key: bool
```
Comment (ypise) [inline api/python/provisioner/api.py:431]: Could you please be specific what is the difference between:
gpg_pub_key
import_pub_key

Should it help if we assume that `import_pub_key`, if specified, also means `import_pub_key` to be true.

**Review by ypise — APPROVED** [review]

File: api/python/provisioner/api.py (hunk 1 above)
Comment (dmitrydb) [reply api/python/provisioner/api.py:431]: Option `gpg_pub_key` is needed more for testing purposes.  It means that by default we assume that Seagate's GPG public key is imported. I provided the `import_pub_key` option to import provided `gpt_pub_key` if needed. Yes, it is correct: if `--import-pub-key` is specified, it means that `import_pub_key` is set to `True`

File: api/python/provisioner/commands/upgrade/set_swupgrade_repo.py [hunk 2]
```diff
@@ -272,6 +273,25 @@ def _pre_repo_validation(self, params: inputs.SWUpgradeRepo,
 
         """
         if not params.is_remote():
+            if params.sig_file:
+                logger.info("File with ISO signature is specified. Start GPG "
+                            "signature validation for the ISO")
+                auth_validator = AuthenticityValidator(
```
Comment (andkononykhin2) [inline api/python/provisioner/commands/upgrade/set_swupgrade_repo.py:279]: I suggest to consider to call `CheckISOAuthenticity` here instead.

**Review by andkononykhin2 — CHANGES_REQUESTED** [review]
@dmitrydb Looks good. But we need to change some inputs since current UX is going to be a bit different. Please check my comments. Thank you

File: api/python/provisioner/commands/upgrade/set_swupgrade_repo.py (hunk 2 above)
Comment (dmitrydb) [reply api/python/provisioner/commands/upgrade/set_swupgrade_repo.py]: Done

File: api/python/provisioner/inputs.py [hunk 3]
```diff
@@ -1144,6 +1144,16 @@ class SWUpgradeRepo(SWUpdateRepo):
         default=None,
         converter=lambda x: x and config.HashType(str(x))
     )
+    sig_file: Optional[Union[str, Path]] = attr.ib(
```
[the remark this answers is not in the document]
Comment (dmitrydb) [reply api/python/provisioner/inputs.py:1147]: Done

File: api/python/provisioner/config.py [hunk 4]
```diff
@@ -608,7 +608,13 @@ class HareStatus(Enum):
 
     FAILED = 'Failed'
     SUCCEEDED = 'Succeeded'
-    IN_PROGRESS = 'InProgress'
```
[the remark this answers is not in the document]
Comment (dmitrydb) [reply api/python/provisioner/config.py:611]: It was deleted by mistake: I tried to reuse this enum for ISO signature validation purposes and not fully revert my changes. I revert it back.

File: api/python/provisioner/api.py [hunk 5]
```diff
@@ -414,6 +414,34 @@ def get_swupgrade_info(release: str = None, iso_path: str = None,
                      nowait=nowait)
 
 
+def check_iso_authenticity(iso_path: str, sig_file: str,
+                           gpg_pub_key: str = None,
+                           import_pub_key: bool = False, nowait=False):
```
[the remark this answers is not in the document]
Comment (dmitrydb) [reply api/python/provisioner/api.py:419]: Done

File: api/python/provisioner/config.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (andkononykhin2) [reply api/python/provisioner/config.py:611]: thanks

**Review by andkononykhin2 — APPROVED** [review]

**Pull request merged**